File: public/examples/integrations/mcp-servers/brightdata/search_engine_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "Brightdata.SearchEngine";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "query": "best noise cancelling headphones 2025",
  "engine": "google",
  "language": "en",
  "country_code": "us",
  "search_type": "shopping",
  "start": 0,
  "num_results": 5,
  "location": "San Francisco, CA",
  "device": "android",
  "return_json": true
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));

File: public/examples/integrations/mcp-servers/brightdata/search_engine_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "Brightdata.SearchEngine"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'query': 'best noise cancelling headphones 2025',
    'engine': 'google',
    'language': 'en',
    'country_code': 'us',
    'search_type': 'shopping',
    'start': 0,
    'num_results': 5,
    'location': 'San Francisco, CA',
    'device': 'android',
    'return_json': True
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))
